extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/transit/SigningResult.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.vault.runtime.transit;
+
+public class SigningResult extends VaultTransitBatchResult<String> {
+
+    // ed25519
+    private String publicKey;
```

Comment:
How can returning the public key help ? The signing key pairs are managed by Vault, and even though it is a public one, still not sure it needs to be returned

Reply:
it is part of the vault api. so I assumed this had value. 
I actually do not know why it would make sense to return the public key for `ed25519` and not for the other asymmetric key types.
from the doc, it sounds like `ed25519`stands apart from the other algorithms:

> ED25519 (asymmetric, supports derivation). When using derivation, a sign operation with the same context will derive the same key and signature; this is a signing analogue to convergent_encryption
...
> hash_algorithm (string: "sha2-256") – Specifies the hash algorithm to use for supporting key types (notably, not including ed25519 which specifies its own hash algorithm).
...
> context (string: "") - Base64 encoded context for key derivation. Required if key derivation is enabled; currently only available with ed25519 keys.
...
> Given an ed25519 key with derived keys set, the context parameter is expected for each batch_input item, and the response will include the derived public key for each item.

by the way, `SigningResult` is a runtime class. it is not part of the api, since we changed the `sign` signature to return a `Map<SigningRequest, String>`

so I am going to leave it in the dto class `VaultTransitSignDataBatchResult`, but will remove it from the `SigningResult`.